thrimbletrimmer/dashboard.html

```diff
@@ -0,0 +1,247 @@
+                document.getElementById('StateFilter').appendChild(row);
+            });
+            return events;
+        }
+
+        function filterEvents(events) {
+            if(!window.location.search) { return events; }
+
+            var urlParams = new URLSearchParams(window.location.search);
+
+            if(urlParams.has("start")) {
+                let startDate = new Date(urlParams.get("start"));
+                events = events.filter(event => new Date(event.event_start) >= startDate);
+            }
+
+            if(urlParams.has("end")) {
+                let endDate = new Date(urlParams.get("end"));
+                events = events.filter(event => new Date(event.event_start) <= endDate);
+            }
+
+            if(urlParams.has("category")) {
+                events = events.filter(event => urlParams.get("category").split(",").indexOf(event.category) >= 0);
+            }
+
+            if(urlParams.has("state")) {
+                events = events.filter(event => urlParams.get("state").split(",").indexOf(event.state) >= 0);
+            }
+
+            return events;
+        }
+
+        function populateTable (events) {
+            events.forEach(event => {
+                let row = document.createElement("TR");
+                row.innerHTML = `
+                    <td>${event.event_start}</td>
+                    <td>${event.event_end}</td>
+                    <td class="${event.category.replace(/ /g, "")}">${event.category}</td>
+                    <td>${event.description}</td>
+                    <td>${event.state}</td>
+                    <td><a href="/thrimbletrimmer?id=${event.id}">Edit</a></td>
+                    <td>${event.video_link ? "<a href='"+event.video_link+"'>"+event.video_link+"</a>":""}</td>
+                `;
+                document.getElementById('QueueTable').appendChild(row);
+            });
+        }
+
+        function applyFilters() {
+            let startDateFilter = document.getElementById("DateFilterStart").value;
+            let endDateFilter = document.getElementById("DateFilterEnd").value;
+            let categoryfilters = Array.from(document.querySelectorAll("#CategoryFilter :checked")).map(checkbox => checkbox.getAttribute("filtervalue")).join();
+            let statefilters = Array.from(document.querySelectorAll("#StateFilter :checked")).map(checkbox => checkbox.getAttribute("filtervalue")).join();
+            window.location.href = `/thrimbletrimmer/dashboard.html?${startDateFilter ? "start="+startDateFilter+"&":""}${endDateFilter ? "end="+endDateFilter+"&":""}category=${categoryfilters}&state=${statefilters}`;
+        }
+
+        //On Page Load
+        getEvents().then(populateFilters).then(filterEvents).then(populateTable);
+    </script>
+</body>
+</html>
```

Comment:
please ensure your files end in a newline, as per unix conventions. otherwise various bits of tooling may not handle it correctly.